pkgs/development/libraries/qt-6/default.nix

```diff
@@ -42,7 +42,7 @@ let
       qtModule = callPackage ./qtModule.nix { };
 
       qtbase = callPackage ./modules/qtbase.nix {
-        withGtk3 = true;
+        withGtk3 = !stdenv.hostPlatform.isMinGW;
```

Comment:
Isn't gtk3 available on mingw?

Reply:
Nope
``` 
nix build '.#pkgs.pkgsCross.mingw32.gtk3'

error:
       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'gtk+3-i686-w64-mingw32-3.24.39'
         whose name attribute is located at /nix/store/kp890n9higzcq3n6bzqimvnl2qkx5i1v-source/pkgs/stdenv/generic/make-derivation.nix:353:7

       … while evaluating attribute 'buildInputs' of derivation 'gtk+3-i686-w64-mingw32-3.24.39'

         at /nix/store/kp890n9higzcq3n6bzqimvnl2qkx5i1v-source/pkgs/stdenv/generic/make-derivation.nix:400:7:

          399|       depsHostHost                = elemAt (elemAt dependencies 1) 0;
          400|       buildInputs                 = elemAt (elemAt dependencies 1) 1;
             |       ^
          401|       depsTargetTarget            = elemAt (elemAt dependencies 2) 0;

       error: Package ‘libxkbcommon-1.5.0’ in /nix/store/kp890n9higzcq3n6bzqimvnl2qkx5i1v-source/pkgs/development/libraries/libxkbcommon/default.nix:53 is not available on the requested hostPlatform:
         hostPlatform.config = "i686-w64-mingw32"
         package.meta.platforms = [
           "i686-cygwin"
           "x86_64-cygwin"
           "x86_64-darwin"
           "i686-darwin"
           "aarch64-darwin"
           "armv7a-darwin"
           "i686-freebsd13"
           "x86_64-freebsd13"
           "x86_64-solaris"
           "aarch64-linux"
           "armv5tel-linux"
           "armv6l-linux"
           "armv7a-linux"
           "armv7l-linux"
           "i686-linux"
           "loongarch64-linux"
           "m68k-linux"
           "microblaze-linux"
           "microblazeel-linux"
           "mips-linux"
           "mips64-linux"
           "mips64el-linux"
           "mipsel-linux"
           "powerpc64-linux"
           "powerpc64le-linux"
           "riscv32-linux"
           "riscv64-linux"
           "s390-linux"
           "s390x-linux"
           "x86_64-linux"
           "aarch64-netbsd"
           "armv6l-netbsd"
           "armv7a-netbsd"
           "armv7l-netbsd"
           "i686-netbsd"
           "m68k-netbsd"
           "mipsel-netbsd"
           "powerpc-netbsd"
           "riscv32-netbsd"
           "riscv64-netbsd"
           "x86_64-netbsd"
           "i686-openbsd"
           "x86_64-openbsd"
           "x86_64-redox"
         ]
         package.meta.badPlatforms = [ ]
       , refusing to evaluate.

       a) To temporarily allow packages that are unsupported for this system, you can use an environment variable
          for a single invocation of the nix tools.

            $ export NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM=1

          Note: When using `nix shell`, `nix build`, `nix develop`, etc with a flake,
                then pass `--impure` in order to allow use of environment variables.

       b) For `nixos-rebuild` you can set
         { nixpkgs.config.allowUnsupportedSystem = true; }
       in configuration.nix to override this.

       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
         { allowUnsupportedSystem = true; }
       to ~/.config/nixpkgs/config.nix.

```

Reply:
Then, using `!lib.meta.availableOn stdenv.hostPlatform gtk3` is more appropriate.

Reply:
I think there is a bug in the in the gtk3 package 
```
nix-repl> lib.meta.availableOn pkgsCross.mingw32.stdenv.hostPlatform gtk3
true


nix-repl> :b pkgsCross.mingw32.gtk3
error:
       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'gtk+3-i686-w64-mingw32-3.24.39'
         whose name attribute is located at /home/sean/nixpkgs-contributing/pkgs/stdenv/generic/make-derivation.nix:353:7

       … while evaluating attribute 'buildInputs' of derivation 'gtk+3-i686-w64-mingw32-3.24.39'

         at /home/sean/nixpkgs-contributing/pkgs/stdenv/generic/make-derivation.nix:400:7:

          399|       depsHostHost                = elemAt (elemAt dependencies 1) 0;
          400|       buildInputs                 = elemAt (elemAt dependencies 1) 1;
             |       ^
          401|       depsTargetTarget            = elemAt (elemAt dependencies 2) 0;

       error: Package ‘libxkbcommon-1.5.0’ in /home/sean/nixpkgs-contributing/pkgs/development/libraries/libxkbcommon/default.nix:60 is not available on the requested hostPlatform:
         hostPlatform.config = "i686-w64-mingw32"
         package.meta.platforms = [
           "i686-cygwin"
           "x86_64-cygwin"
           "x86_64-darwin"
           "i686-darwin"
           "aarch64-darwin"
           "armv7a-darwin"
           "i686-freebsd13"
           "x86_64-freebsd13"
           "x86_64-solaris"
           "aarch64-linux"
           "armv5tel-linux"
           "armv6l-linux"
           "armv7a-linux"
           "armv7l-linux"
           "i686-linux"
           "loongarch64-linux"
           "m68k-linux"
           "microblaze-linux"
           "microblazeel-linux"
           "mips-linux"
           "mips64-linux"
           "mips64el-linux"
           "mipsel-linux"
           "powerpc64-linux"
           "powerpc64le-linux"
           "riscv32-linux"
           "riscv64-linux"
           "s390-linux"
           "s390x-linux"
           "x86_64-linux"
           "aarch64-netbsd"
           "armv6l-netbsd"
           "armv7a-netbsd"
           "armv7l-netbsd"
           "i686-netbsd"
           "m68k-netbsd"
           "mipsel-netbsd"
           "powerpc-netbsd"
           "riscv32-netbsd"
           "riscv64-netbsd"
           "x86_64-netbsd"
           "i686-openbsd"
           "x86_64-openbsd"
           "x86_64-redox"
         ]
         package.meta.badPlatforms = [ ]
       , refusing to evaluate.

       a) To temporarily allow packages that are unsupported for this system, you can use an environment variable
          for a single invocation of the nix tools.

            $ export NIXPKGS_ALLOW_UNSUPPORTED_SYSTEM=1

          Note: When using `nix shell`, `nix build`, `nix develop`, etc with a flake,
                then pass `--impure` in order to allow use of environment variables.

       b) For `nixos-rebuild` you can set
         { nixpkgs.config.allowUnsupportedSystem = true; }
       in configuration.nix to override this.

       c) For `nix-env`, `nix-build`, `nix-shell` or any other Nix command you can add
         { allowUnsupportedSystem = true; }
       to ~/.config/nixpkgs/config.nix.

```

Reply:
No change since gtk3 appears to indicate it has mingw support when in fact it doesn't.

Reply:
We really should fix this on the gtk3 side.

Reply:
My hope is that we can leave the mingw conditional for now to get this functionality. When gtk3 actually has mingw support, we can change that line. I suspect that adding mingw support for gtk3 may not be trivial.